### PR TITLE
[Feature][Connector-V2][Jdbc] Add oceanbase dialect factory

### LINKED ISSUE #2617: [Connector-V2][OceanBase] Support database: OceanBase
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

OceanBase
https://www.oceanbase.com/

Support connect OceanBase https://github.com/apache/incubator-seatunnel/issues/1946
Add OceanBase source & sink code 

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by whhe**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

Implemented based on other 2 forks from the community including #4626, close #2617 
- https://github.com/changhuyan/incubator-seatunnel/commit/cd2171e591bb521b01013da97382672590b60268
- https://github.com/silenceland/incubator-seatunnel/commit/adcd203dd53e8896699561eed539537fc7b0a9b4

### CODE REVIEW

**Comment by whhe** [comment]
Run the oceanbase mysql mode test case with `./mvnw verify -Dtest=JdbcOceanBaseMysqlIT -DfailIfNoTests=false -am` and get the log as below.

```
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 245.631 s - in org.apache.seatunnel.connectors.seatunnel.jdbc.JdbcOceanBaseMysqlIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0
```
[maven.log](https://github.com/apache/seatunnel/files/11903679/maven.log)

File: docs/en/connector-v2/sink/Jdbc.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ support `Xa transactions`. You can set `is_exactly_once=true` to enable it.
 | user                                      | String  | No       | -             |
 | password                                  | String  | No       | -             |
 | query                                     | String  | No       | -             |
+| driver_type                               | String  | No       | -             |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/Jdbc.md]: Hi, Thanks for you contribution. Can you add OceanBase Source and Sink Document like https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/Mysql.md and https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/sink/Mysql.md

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSourceFactory.java [hunk 2]
```diff
@@ -83,7 +83,10 @@ TableSource<T, SplitT, StateT> createSource(TableFactoryContext context) {
         JdbcConnectionProvider connectionProvider =
                 new SimpleJdbcConnectionProvider(config.getJdbcConnectionConfig());
         final String querySql = config.getQuery();
-        JdbcDialect dialect = JdbcDialectLoader.load(config.getJdbcConnectionConfig().getUrl());
+        JdbcDialect dialect =
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSourceFactory.java:87]: Add new option to `public OptionRule optionRule() `

File: docs/en/connector-v2/source/OceanBase.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+| query                        | String | Yes      | -               | Query statement                                                                                                                                                                                                                                                       |
+| connection_check_timeout_sec | Int    | No       | 30              | The time in seconds to wait for the database operation used to validate the connection to complete                                                                                                                                                                    |
+| partition_column             | String | No       | -               | The column name for parallelism's partition, only support numeric type primary key and only support single column.                                                                                                                                                    |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/OceanBase.md]:
```suggestion
| partition_column             | String | No       | -               | The column name for parallelism's partition, only support numeric type column and string type column.                                                                                                                                                    |
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/resources/jdbc_oceanbase_oracle_source_and_sink.conf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      query = "SELECT * FROM source"
+      compatible_mode = "oracle"
+ }
+}
+
+transform {
+}
+
+sink {
+  Assert {
```
Comment (EricJoy2048) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/resources/jdbc_oceanbase_oracle_source_and_sink.conf]: Hi, thank you for your contribution. A good test case should not just check whether the number of rows is correct. You should check the field type of each column in each row and whether the value of the field is completely correct.
You can reference https://github.com/apache/seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-clickhouse-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/ClickhouseIT.java to to it.

Thanks.

**Comment by whhe** [comment]
Enable JdbcOceanBaseMysqlIT and tested with `./mvnw verify -Dtest=JdbcOceanBaseMysqlIT -DfailIfNoTests=false -am`, output: [nohup.log](https://github.com/apache/seatunnel/files/11979200/nohup.log).

```
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 276.756 s - in org.apache.seatunnel.connectors.seatunnel.jdbc.JdbcOceanBaseMysqlIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0
```

File: docs/en/connector-v2/sink/OceanBase.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
+# OceanBase
+
+> JDBC OceanBase Sink Connector
+
+## Support Those Engines
+
+> Spark<br/>
+> Flink<br/>
+> Seatunnel Zeta<br/>
```
Comment (Hisoka-X) [inline docs/en/connector-v2/sink/OceanBase.md]:
```suggestion
> SeaTunnel Zeta<br/>
```

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialectFactory.java [hunk 6]
```diff
@@ -37,4 +37,14 @@ public interface JdbcDialectFactory {
 
     /** @return Creates a new instance of the {@link JdbcDialect}. */
     JdbcDialect create();
+
+    /**
+     * Create a {@link JdbcDialect} instance based on the driver type.
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialectFactory.java]:
```suggestion
     * Create a {@link JdbcDialect} instance based on the driver type and compatible mode.
```

**Review by Hisoka-X — DISMISSED** [review]
LGTM, except some comment.

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**